public/scripts/prepare-create-assistant.js

```diff
@@ -0,0 +1,84 @@
+import { t } from './i18n.js';
+import { SlashCommandParser } from './slash-commands/SlashCommandParser.js';
+import { getContext } from './st-context.js';
+
+const ASSISTANT_NAME = 'Assistant';
+
+/**
+ * @desc Prepare the create assistant button.
+ * @returns {void}
+ */
+export function prepareCreateAssistant() {
+    const createAssistantBtn = document.querySelector('[data-which="create-assistant-btn"]');
+
+    if (createAssistantBtn) {
+        createAssistantBtn.addEventListener('click', async () => {
+            if (assistantAlreadyExists()) {
+                chatWithAssistant();
+            } else {
+                await createAssistant();
+                chatWithAssistant();
+            }
+        });
+    }
+}
+
+// region Helpers
+/**
+ * @desc Sleep for a given number of milliseconds.
+ * @param {number} ms
+ * @returns {Promise<void>}
+ */
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
```

Comment:
Are you sure you need to do that weird stuff with sleeping? Should not be the case. for the UI interactions.

If you really need to, we already have `delay()` function exported which does the same.